File: src/app/models/response.ts
export enum ResponseStatus {
  SUCCESS = 'success',
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error',
}

export enum ResponseStatusIcon {
  SUCCESS = 'check_circle',
  INFO = 'info_outlined',
  WARNING = 'warning',
  ERROR = 'error',
  '' = 'notifications',
}

export interface INotify {
  status: ResponseStatus;
  message: string;
}

export interface ActiveNotification extends INotify {
  id: number;
  width: number;
  color: string;
  icon: ResponseStatusIcon;
  intervalId?: number;
  timeoutId?: number;
}

export class Response {
  constructor(
    public status: ResponseStatus,
    public message: string,
    public data: any
  ) {}
}

File: src-tauri/src/models/response.rs
/* sys lib */
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ResponseStatus {
  Success,
  Info,
  Warning,
  Error,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum DataValue {
  String(String),
  Number(f64),
  Bool(bool),
  Array(Vec<serde_json::Value>),
  Object(serde_json::Value),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ResponseModel {
  pub status: ResponseStatus,
  pub message: String,
  pub data: DataValue,
}
